motionplan/nearestNeighbor.go

```diff
@@ -111,24 +106,15 @@ func (nm *neighborManager) parallelNearestNeighbor(
 	for k := range rrtMap {
 		nm.nnKeys <- k
 	}
-	nm.nnLock.Lock()
-	nm.ready = true
-	nm.nnLock.Unlock()
+	close(nm.nnKeys)
+
 	var best node
 	bestDist := math.Inf(1)
-	returned := 0
-	for returned < nm.nCPU {
-		select {
```

Comment:
I didn't see the purpose of having a select statement here. Specifically, the `default` case would cause this goroutine to busy-loop when blocking seems the desired behavior.